common/metrics/config.go

```diff
@@ -130,6 +130,8 @@ const (
 	FrameworkTally = "tally"
 	// FrameworkOpentelemetry OpenTelemetry framework id
 	FrameworkOpentelemetry = "opentelemetry"
+	// FrameworkCustom Custom framework id
+	FrameworkCustom = "custom"
```

Comment:
I wouldn't add this. Just completely ignore config if custom reporter is not `nil` in server options.